### ISSUE TITLE
dotty scalajs toplevel object import compilation failure

### ISSUE THREAD

**Issue by aappddeevv** [lampepfl/dotty#9755]
Reference: https://github.com/scala-js/scala-js/issues/4183

## Minimized code

```scala
package test1
//...imports
@js.native
@JSImport("node-fetch", JSImport.Default)
def fetch2(url: String, arg: js.Dynamic): js.Promise[Response] = js.native
```

## Output
fails with 
```sh
[error] 19 |def fetch2(url: String, arg: js.Dynamic): js.Promise[Response] = js.native
[error]    |                                                                 ^^^^^^^^^
[error]    |       js.native may only be used as stub implementation in facade types
[error] one error found
[error] (Compile / compileIncremental) Compilation failed
```

## Expectation
It should compile.

## Note
However, the more traditional way of writing it works:
```scala
@js.native
@JSImport("node-fetch", JSImport.Default)
object fetch extends js.Any:
    def apply(url: String, arg: js.Dynamic): js.Promise[Response] = js.native
```

**Comment by sjrd** [comment]
The title mentions a package object, but the snippet seems to use a top-level definition instead. Which one is it?

**Comment by aappddeevv** [comment]
It's toplevel. I'm still getting used to the vocab change.

**Comment by aappddeevv** [comment]
There may be other toplevel issues. I'm seeing a problem where a `@JSImport` at the top level is not emitting a require/import js clause in the output as well. 

I'll open that separately once I confirm that its a toplevel issue, if it is.

**Comment by aappddeevv** [comment]
It looks like #9725 did not address this issue. You did not claim that it did of course since this is an older PR, but would you have expected this to be fixed by #9725?

**Comment by sjrd** [comment]
Not necessarily. I had not tested with top-level vals and defs yet.
